pkg/xds/envoy/clusters/health_check_configurer.go

```diff
@@ -25,9 +25,32 @@ func (e *healthCheckConfigurer) Configure(cluster *envoy_api.Cluster) error {
 		return nil
 	}
 	activeChecks := e.healthCheck.Spec.Conf
+	tcpHealthCheck := &envoy_core.HealthCheck_TcpHealthCheck{}
+
+	if activeChecks.Tcp != nil {
+		if activeChecks.Tcp.Send != nil {
+			tcpHealthCheck.Send = &envoy_core.HealthCheck_Payload{
+				Payload: &envoy_core.HealthCheck_Payload_Binary{
```

Comment:
Are you sure it works?
https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/core/health_check.proto#envoy-api-msg-core-healthcheck-payload

it's missing from docs and here it says it's not implemented

https://github.com/envoyproxy/envoy/blob/230d1ec32c864a657361441f9c3723d6e2af843b/api/envoy/api/v2/core/health_check.proto#L68

same with V3
https://github.com/envoyproxy/envoy/blob/230d1ec32c864a657361441f9c3723d6e2af843b/api/envoy/config/core/v3/health_check.proto#L61

Reply:
On checkin it right now.

Reply:
Tested it and it was working correctly, but as you mentioned there are notes it shouldn't be implemented, I rewrote it to suggested `Text` versions.